chore(multi-region): add operational simplification from Zeebe 8.6. (#4205)
camunda/camunda-docs: docs/self-managed/operational-guides/multi-region/dual-region-ops.md

:::

> :::info
> 
> The operational procedure was reworked in Camunda 8.6 to be more straightforward and reliable.
> 
> The old procedure of 8.5 [as outlined](../../../../versioned_docs/version-8.5/self-managed/operational-guides/multi-region/dual-region-ops.md) is still compatible with Camunda 8.6 but compatibility will be removed in 8.7.
> 
> :::
> 

Reviewer: ```suggestion
:::info
This procedure has been updated in the Camunda 8.6 release. The procedure used in Camunda 8.5 has been deprecated, and compatibility will be removed in the 8.7 release.
:::
```
I'm so-so on if you'd like to include the link to the old version. If we're deprecating that method, I don't necessarily see the value in it, unless there are circumstances where the older procedure would be required. 

Can we move this up to the top of the page as well, for visibility?

Author: I'm fine not including the link, a person can just use the navbar on top to quickly switch the version.
It was just for people to have a grace period to readjust their own documentation.

Sure moved it to the top.

#### Desired state

You have removed the lost brokers from the Zeebe cluster. This will allow us to continue processing after the next step and ensure that the new brokers in the failback procedure will only join the cluster with our intervention.

#### How to get there

You will port-forward the `Zeebe Gateway` in the surviving region to the local host to interact with the Gateway.

The following alternatives to port-forwarding are possible:

- if it's exposed to the outside, one can skip port-forwarding and use the URL directly

> - one can use, for example, the Elasticsearch pod to exec into it and curl from there

Reviewer: ```suggestion
- one can `exec` into an existing pod (such as Elasticsearch), and `curl` from there
```

^ Is this a correct rephrasing? Some clarity + command backtics.

Author: sure, I also linked to the official k8s exec docs on the command.

Gateway version: 8.6.0
Brokers:
  Broker 0 - camunda-zeebe-0.camunda-zeebe.camunda-london.svc:26501
    Version: 8.6.0
    Partition 1 : Leader, Healthy
    Partition 6 : Follower, Healthy
    Partition 7 : Follower, Healthy
    Partition 8 : Follower, Healthy
  Broker 2 - camunda-zeebe-1.camunda-zeebe.camunda-london.svc:26501
    Version: 8.6.0
    Partition 1 : Follower, Healthy
    Partition 2 : Follower, Healthy
    Partition 3 : Follower, Healthy
    Partition 8 : Leader, Healthy
  Broker 4 - camunda-zeebe-2.camunda-zeebe.camunda-london.svc:26501
    Version: 8.6.0
    Partition 2 : Follower, Healthy
    Partition 3 : Leader, Healthy
    Partition 4 : Follower, Healthy
    Partition 5 : Follower, Healthy
  Broker 6 - camunda-zeebe-3.camunda-zeebe.camunda-london.svc:26501
    Version: 8.6.0
    Partition 4 : Follower, Healthy
    Partition 5 : Follower, Healthy
    Partition 6 : Follower, Healthy
    Partition 7 : Leader, Healthy
```

  </summary>
</details>

2. Portforward the service of the Zeebe Gateway for the REST API

```bash
kubectl --context $CLUSTER_SURVIVING port-forward services/$HELM_RELEASE_NAME-zeebe-gateway 9600:9600 -n $CAMUNDA_NAMESPACE_SURVIVING
```

> 3. Based on the [Cluster Scaling APIs](../../zeebe-deployment/operations/cluster-scaling.md) you will send a request to the Zeebe Gateway to redistribute the load to the remaining brokers, thereby removing the lost brokers.

Reviewer: ```suggestion
3. Based on the [Cluster Scaling APIs](../../zeebe-deployment/operations/cluster-scaling.md), send a request to the Zeebe Gateway to redistribute the load to the remaining brokers, thereby removing the lost brokers.
```

Zeebe is not yet be able to continue exporting data since the Zeebe brokers in the surviving region are configured to point to the Elasticsearch instance of the lost region.

#### Desired state

You have disabled the Elasticsearch exporter to the failed region in the Zeebe cluster.

The Zeebe cluster is then unblocked and can export data to Elasticsearch again.

Completing this step will restore regular interaction with Camunda 8 for your users, marking the conclusion of the temporary recovery.

#### How to get there

1. Portforward the service of the Zeebe Gateway for the REST API

```bash
kubectl --context $CLUSTER_SURVIVING port-forward services/$HELM_RELEASE_NAME-zeebe-gateway 9600:9600 -n $CAMUNDA_NAMESPACE_SURVIVING
```

2. List all exporters to find the corresponding ID. Alternatively, you can check your `camunda-values.yml` file, which lists the exporters as those had to be configured explicitly.

```bash

> curl -XGET 'http://localhost:9600/actuator/exporters'

Author: great! I left a todo in the text where we should probably link it and only have a wrong placeholder atm.

# camunda-tasklist   0/0     0            0           23m
```

> For the Zeebe Elasticsearch exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful disabling. This is a synchronous operation.

Reviewer 2: You can also link the api docs above http://localhost:3000/docs/next/self-managed/zeebe-deployment/operations/management-api/#exporting-api

Author: linked it where it's executed.

#### Current state

Camunda 8 is reachable to the end-user but not yet exporting any data.

#### Desired state

You are reactivating the existing exporter and initializing a new exporter to the recreated region. This will allow Zeebe to export data to Elasticsearch again.

#### How to get there

1. Initialize the new exporter for the recreated region by sending an API request via the Zeebe Gateway:

```bash
kubectl --context $CLUSTER_SURVIVING port-forward services/$HELM_RELEASE_NAME-zeebe-gateway 9600:9600 -n $CAMUNDA_NAMESPACE_SURVIVING
curl -XPOST 'http://localhost:9600/actuator/exporters/elasticsearchregion1/enable' -H 'Content-Type: application/json' -d '{"initializeFrom" : "elasticsearchregion0"}'
```

> 2. Reactivate the exporters by sending the API activation request via the Zeebe Gateway:

Reviewer 2: It would be better to split this into a new step. We should resume only after the exporter has enabled. Enabling exporter usually finishes faster, but it is not a synchronous operation. So the process should be as follows.

1. enable exporter
2. verify it is enabled
3. Resume exporting

#### Verification

Port-forwarding the Zeebe Gateway via `kubectl` for the REST API and listing all exporters will reveal their current status.

```bash
kubectl --context $CLUSTER_SURVIVING port-forward services/$HELM_RELEASE_NAME-zeebe-gateway 9600:9600 -n $CAMUNDA_NAMESPACE_SURVIVING
curl -XGET 'http://localhost:9600/actuator/exporters'
```

<details>
  <summary>Example output</summary>
  <summary>

```bash
[{"exporterId":"elasticsearchregion0","status":"ENABLED"},{"exporterId":"elasticsearchregion1","status":"ENABLED"}]
```

  </summary>
</details>

Via the already port-forwarded Zeebe Gateway, you can also check the status of the change by using the Cluster API.

```bash
curl -XGET 'http://localhost:9600/actuator/cluster' | jq .lastChange
```

<details>
  <summary>Example output</summary>
  <summary>

```bash
{
  "id": 6,
  "status": "COMPLETED",
  "startedAt": "2024-08-23T12:54:07.968549269Z",
  "completedAt": "2024-08-23T12:54:09.282558853Z"
}
```

  </summary>
</details>

> For the Zeebe Elasticsearch exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful resumption. This is a synchronous operation.

Reviewer 2: ```suggestion
For the reactivating the exporters, there's currently no API available to confirm this. Only the response code of `204` indicates a successful resumption. This is a synchronous operation.
```
